### ISSUE TITLE
Judgement of the Crusader is set a prio 2 whereas Judgement of Justice/Light/Wisdom are set at prio 3

### ISSUE THREAD

**Issue by theloras** [LightsHope/issues#606]
Given the fact that Paladin Judgements are so unique, there should be parity between ALL Judgements as far as the debuff priority list goes seeing as 3/4 are set at prio 3 whereas Judgement of the Crusader is only set at prio 2:

        case SPELLFAMILY_PALADIN:
        {
            switch (m_spellProto->Id)
            {
                // Judgement of Wisdom (3 ranks)
                case 20186:
                case 20354:
                case 20355:
                // Judgement of Light
                case 20185: // r1
                case 20344: // r2
                case 20345: // r3
                case 20346: // r4
                    m_debuffLimitScore = 3;
                    return;
                // Judgement of the Crusader (6 ranks)
                case 21183:
                case 20188:
                case 20300:
                case 20301:
                case 20302:
                case 20303:
                    m_debuffLimitScore = 2;
                    return;

and then further down for Judgement of Justice (Curse of Recklessness has the same effect):

 case SPELL_AURA_PREVENTS_FLEEING:               // Curse of recklessness
currEffectScore = 3;

https://github.com/lh-server/core/blob/development/src/game/Spells/SpellAuras.cpp#L6815

**Comment by theloras** [comment]
the great irony is that neither JoL or JoW are listed in the overall debuff prio list:

 case SPELL_AURA_MOD_TAUNT:
            case SPELL_AURA_MOD_THREAT:
            case SPELL_AURA_MOD_CHARM:
            case SPELL_AURA_MOD_FEAR:
            case SPELL_AURA_MOD_CONFUSE:
            case SPELL_AURA_MOD_POSSESS:
            case SPELL_AURA_MOD_STUN:
                currEffectScore = 4;
                break;
            case SPELL_AURA_MOD_RESISTANCE:
            case SPELL_AURA_MOD_LANGUAGE:                   // Language curse
            case SPELL_AURA_MOD_STALKED:                    // Hunter mark
            case SPELL_AURA_MOD_DISARM:
            case SPELL_AURA_MOD_DAMAGE_PERCENT_TAKEN:
            case SPELL_AURA_PREVENTS_FLEEING:               // Curse of recklessness
            case SPELL_AURA_MOD_ATTACKER_SPELL_CRIT_CHANCE: // Winter's Chill
            case SPELL_AURA_MOD_MELEE_HASTE:
            case SPELL_AURA_MOD_ATTACK_POWER:
            case SPELL_AURA_MOD_DAMAGE_DONE:
            **case SPELL_AURA_MOD_DAMAGE_TAKEN:**
            case SPELL_AURA_MOD_HEALING_PCT:                // Mortal Strike
                **currEffectScore = 3;**

but Judgement of the Crusader's functionality is:

Judgement of the Crusader | Rank 6
Effect #1 **(6) Apply Aura #14: Mod Damage Taken (2)** 
               Value: 140
http://classicdb.ch/?spell=20303

yet it was set at prio 2 while JoL and JoW were set at 3:

Judgement of Wisdom
Rank 3
100 yd range
Instant
Effect #1    (30) Energize 
Value: 59
http://classicdb.ch/?spell=20353

Judgement of Light
Rank 4
100 yd range
Instant
Effect #1    (10) Heal 
Value: 61
http://classicdb.ch/?spell=20343

**Comment by Gamemechanicwow** [comment]
> the great irony is that neither JoL or JoW are listed in the overall debuff prio list:
> 
> case SPELL_AURA_MOD_TAUNT:
> case SPELL_AURA_MOD_THREAT:
> case SPELL_AURA_MOD_CHARM:
> case SPELL_AURA_MOD_FEAR:
> case SPELL_AURA_MOD_CONFUSE:
> case SPELL_AURA_MOD_POSSESS:
> case SPELL_AURA_MOD_STUN:
> currEffectScore = 4;
> break;
> case SPELL_AURA_MOD_RESISTANCE:
> case SPELL_AURA_MOD_LANGUAGE: // Language curse
> case SPELL_AURA_MOD_STALKED: // Hunter mark
> case SPELL_AURA_MOD_DISARM:
> case SPELL_AURA_MOD_DAMAGE_PERCENT_TAKEN:
> case SPELL_AURA_PREVENTS_FLEEING: // Curse of recklessness
> case SPELL_AURA_MOD_ATTACKER_SPELL_CRIT_CHANCE: // Winter's Chill
> case SPELL_AURA_MOD_MELEE_HASTE:
> case SPELL_AURA_MOD_ATTACK_POWER:
> case SPELL_AURA_MOD_DAMAGE_DONE:
> case SPELL_AURA_MOD_DAMAGE_TAKEN:
> case SPELL_AURA_MOD_HEALING_PCT: // Mortal Strike
> currEffectScore = 3;
> 
> but Judgement of the Crusader's functionality is:
> 
> Judgement of the Crusader | Rank 6
> Effect #1 (6) Apply Aura #14: Mod Damage Taken (2)
> Value: 140
> http://classicdb.ch/?spell=20303
> 
> yet it was set at prio 2 while JoL and JoW were set at 3:
> 
> Judgement of Wisdom
> Rank 3
> 100 yd range
> Instant
> Effect #1 (30) Energize
> Value: 59
> http://classicdb.ch/?spell=20353
> 
> Judgement of Light
> Rank 4
> 100 yd range
> Instant
> Effect #1 (10) Heal
> Value: 61
> http://classicdb.ch/?spell=20343

You didn't link the judgements, but the triggering effects from jol/jow. Effects and mod auras are not the same thing. Both jol and jow places a Proc Trigger Spell on the target.

I cant find the irony you see. These spells are most likely handled seperately to avoid bugs by putting a single priority of the group. But most importantly it works!

Personally i think that the retail vanilla debuf prio system was way more complex than we what we currently implemented. Most likely due to lack of credible sources? I have no clue what the priorities on the judgements where, but i doubt that the retail devs had the same thought "Given the fact that Paladin Judgements are so unique, there should be parity between ALL Judgements as far as the debuff priority list goes". I would assume the priorities was based on raid gamemechanics.

**Comment by theloras** [comment]
The irony is that JoL/JoW were manually set to prio 3 when their functionality wasn't listed in the core of prio 3 debuffs, yet JotC was manually set to prio 2 even though JotC's literal functionality is listed in the main core of prio 3 debuffs:

**case SPELL_AURA_MOD_DAMAGE_TAKEN:
currEffectScore = 3;**

Judgement of the Crusader | Rank 6
**Effect #1 (6) Apply Aura #14: Mod Damage Taken (2)**
Value: 140
http://classicdb.ch/?spell=20303

Now compare JotC with these:

Gift of Arthas
Melee Range 
Instant
Increases physical damage taken by 8 for 3 min
**Effect #1	(6) Apply Aura #14: Mod Damage Taken (1)** 
Value: 8
http://classicdb.ch/?spell=11374

Hemorrhage
Rank 3
35 energy
Melee Range 
Instant
An instant strike that damages the opponent and causes the target to hemorrhage, increasing any Physical damage dealt to the target by up to 7. Lasts 30 charges or 15 sec. Awards 1 combo point.
**Effect #3	(6) Apply Aura #14: Mod Damage Taken (1)** 
Value: 7

JotC is set to prio 2 but Gift of Arthas, Hemo and other type debuffs are set to prio 3?!?!?

Likewise, there are no links for the actual Judgements as they are part of the SEALS:

**Seal of Wisdom**
Rank 3
200 mana
Instant
Fills the Paladin with divine wisdom for 30 sec, giving each melee attack a chance to restore 90 of the Paladin's mana. Only one Seal can be active on the Paladin at any one time.

**Unleashing this Seal's energy will judge an enemy for 10 sec, granting attacks and spells used against the judged enemy a chance to restore 59 mana to the attacker. Your melee strikes will refresh the spell's duration. Only one Judgement per Paladin can be active at any one time.**
http://classicdb.ch/?spell=20357

**Seal of Light**
Rank 4
210 mana
Instant
Fills the Paladin with divine light for 30 sec, giving each melee attack a chance to heal the Paladin for 94. Only one Seal can be active on the Paladin at any one time.

**Unleashing this Seal's energy will judge an enemy for 10 sec, granting melee attacks made against the judged enemy a chance of healing the attacker for 61. Your melee strikes will refresh the spell's duration. Only one Judgement per Paladin can be active at any one time.**
http://classicdb.ch/?spell=20349

**Seal of Justice**
13% of base mana
Instant
Fills the Paladin with the spirit of justice for 30 sec, giving each melee attack a chance to stun for 2 sec. Only one Seal can be active on the Paladin at any one time.

**Unleashing this Seal's energy will judge an enemy for 10 sec, preventing them from fleeing. Your melee strikes will refresh the spell's duration. Only one Judgement per Paladin can be active at any one time.**
http://classicdb.ch/?spell=20164

**Seal of the Crusader**
Rank 6
160 mana
Instant
Fills the Paladin with the spirit of a crusader for 30 sec, granting 306 melee attack power. The Paladin also attacks 40% faster, but deals less damage with each attack. Only one Seal can be active on the Paladin at any one time.

**Unleashing this Seal's energy will judge an enemy for 10 sec, increasing Holy damage taken by up to 140. Your melee strikes will refresh the spell's duration. Only one Judgement per Paladin can be active at any one time.**
http://classicdb.ch/?spell=20308

TLDR - if anything, JotC should be a prio 3 debuff for the simple fact as to be on parity with EVERY OTHER DEBUFF THAT HAS case SPELL_AURA_MOD_DAMAGE_TAKEN:

**Comment by theloras** [comment]
If you want, we can go back even further to this beauty from BETA upon which Judgement of the Crusader is based on:

**Crusader Strike**
Rank 5
90 mana
Melee Range 
Instant
A strike that causes 103 damage and increases the holy damage taken by the target by up to 30 per Crusader Strike. Can be applied up to 5 times. Lasts 30 sec.
**Effect #2	(6) Apply Aura #14: Mod Damage Taken (2)** 
Value: 30
http://classicdb.ch/?spell=10337

But to repeat, separate and player cast/activated spell IDs for the various Judgements never existed in Vanilla - they were baked into the actual Seals themselves which were then activated by this:

**Judgement**
6% of base mana
10 yd range
Instant	10 sec cooldown
**Unleashes the energy of a Seal spell upon an enemy. Refer to individual Seals for Judgement effect.**
Effect #1	(77) Script Effect 
Value: 0
http://classicdb.ch/?spell=20271

It was not until Wrath of the Lich King that Seals and their respective Judgements were separated from one another on top of Seal/Judgement of the Crusader being removed from the game completely and its inherent +Holy Damage baked into our damage Seals and Judgements.

**Comment by Gamemechanicwow** [comment]
>JotC is set to prio 2 but Gift of Arthas, Hemo and other type debuffs are set to prio 3?!?!?

Gift of arthas and hemo are phys damage taken.

> Likewise, there are no links for the actual Judgements as they are part of the SEALS:

I'll try to mansplain it.. I was in no way implying that specific judgements was in vanilla.
You linked the judgement of the crusader debuff, but compared it to the triggering effect from judgements of jol/jow. Which is way of topic, since you wanted to discuss debuffs.

Here is the jol/jow debuffs.
https://classicdb.ch/?spell=20355
https://classicdb.ch/?spell=20346

**Comment by theloras** [comment]
case SPELL_AURA_MOD_RESISTANCE:
case SPELL_AURA_MOD_LANGUAGE: // Language curse
case SPELL_AURA_MOD_STALKED: // Hunter mark
case SPELL_AURA_MOD_DISARM:
case SPELL_AURA_MOD_DAMAGE_PERCENT_TAKEN:
case SPELL_AURA_PREVENTS_FLEEING: // Curse of recklessness
case SPELL_AURA_MOD_ATTACKER_SPELL_CRIT_CHANCE: // Winter's Chill
case SPELL_AURA_MOD_MELEE_HASTE:
case SPELL_AURA_MOD_ATTACK_POWER:
case SPELL_AURA_MOD_DAMAGE_DONE:
case SPELL_AURA_MOD_DAMAGE_TAKEN:
case SPELL_AURA_MOD_HEALING_PCT: // Mortal Strike
currEffectScore = 3;

The Nost Devs hard coded ALL debuffs affecting ALL schools to prio 3 which have this component:

**case SPELL_AURA_MOD_DAMAGE_TAKEN:**

EXCEPT for Judgement of the Crusader which was manually set to prio 2

 // Judgement of the Crusader (6 ranks)
            case 21183:
            case 20188:
            case 20300:
            case 20301:
            case 20302:
            case 20303:
                m_debuffLimitScore = 2;

Whereas they made a manual change to JoL and JoW to make them on the same level as other prio 3 debuffs similar to Judgement of Justice.

I'll try to mansplain it...as far as the justification for this bug fix goes...

1. Give JotC prio parity with the other debuff Judgements

OR

2. Give JotC prio parity with every other debuff with **case SPELL_AURA_MOD_DAMAGE_TAKEN:**

**Comment by Zarant** [comment]
The current implementation is either:
1. Working as intended, 100% based on evidence from 2005-2006
OR
2. An approximation based on whatever written testimonies or anecdotal evidences they managed to find.

If you go with option number 2, you either have to back up your claims with sources dated back from retail vanilla or you can engage into those discussions for days on end. 
Due to the lack of information, most of the server's current systems are just coarse approximations, and sometimes, the core devs fail to properly address game design problems when recreating them, if you think that is the case here, you should further elaborate your argument.
How the code itself was written doesn't really matter, the code is full of inconsistencies and hackfixes, and yet, it's the best approximation we have to date of how stuff worked back in the day.

**Comment by theloras** [comment]
I'm not denying that the code is the best Vanilla coding since actual retail Vanilla - I've played on multiple private servers from the bad old days of Feenix, to Vanilla Gaming.

I just find the arbitrary decision to nerf an entire spell school debuff to prio 2 whereas every other one set at prio 3 to be suspect at best - even though they share the exact same core mechanics:

**case SPELL_AURA_MOD_DAMAGE_TAKEN:**

Doesn't that make anyone else /ponder?

**Comment by Gamemechanicwow** [comment]
> I just find the arbitrary decision to nerf an entire spell school debuff to prio 2 whereas every other one set at prio 3 to be suspect at best - even though they share the exact same core mechanics:
> 
> case SPELL_AURA_MOD_DAMAGE_TAKEN:
> 
> Doesn't that make anyone else /ponder?

Not really. Its different types of damage types. Make a list of every SPELL_AURA_MOD_DAMAGE_TAKEN debuffs and try writing down which classes could benefit from each debuf.

**Comment by theloras** [comment]
On top of Hemorrhage and Gift of Arthas, there is only one other debuff that uses:

**case SPELL_AURA_MOD_DAMAGE_TAKEN:**

**Flame Buffet**
30 yd range
Instant
Inflicts 37 to 44 Fire damage to an enemy and increases the Fire damage it takes by 40 for 20 sec.
Effect #1	(2) School Damage (Fire) 
Value: 37
**Effect #2	(6) Apply Aura #14: Mod Damage Taken (4)** 
Value: 40
http://classicdb.ch/?spell=25668

Which is the debuff that is put up by the Arcanite Dragonling trinket.

So to recap, we have the following debuffs set to prio 3...

- Hemorrhage - that can only be used by a SUBTLETY Rogue
- Gift of Arthas - which has to be made by an Alchemist
- Arcanite Dragonling - that has to be made by an Engineer

Whereas a BASELINE debuff ability for ALL Paladins is MANUALLY override and set to only prio 2?

That just doesn't pass the smell test for me.
